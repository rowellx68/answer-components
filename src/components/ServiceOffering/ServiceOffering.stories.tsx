import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import ServiceOffering, { ServiceOfferingProps } from './ServiceOffering'

export default {
  title: 'Components/ServiceOffering',
  component: ServiceOffering,
  argTypes: {
    className: { control: 'text' },
    title: { control: 'text' },
    text: { control: 'text' },
  },
} as Meta

const Template: Story<ServiceOfferingProps> = (args) => (
  <ServiceOffering {...args} />
)

const GridTemplate: Story<ServiceOfferingProps> = (args) => (
  <div className="grid gap-4 grid-cols-1 md:grid-cols-4">
    <ServiceOffering {...args} />
    <ServiceOffering {...args} />
    <ServiceOffering {...args} />
    <ServiceOffering {...args} />
  </div>
)

export const BasicUsage = Template.bind({})
BasicUsage.args = {
  className: 'pb-6',
  title: 'Service',
  text:
    'This is a really long text that could be used as the sub-heading of the page.',
}

export const GridUsage = GridTemplate.bind({})
GridUsage.args = {
  className: 'pb-6',
  title: 'Services',
  text:
    'This is a really long text that could be used as the sub-heading of the page.',
}
