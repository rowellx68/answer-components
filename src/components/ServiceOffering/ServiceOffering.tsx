import React from 'react'

export interface ServiceOfferingProps {
  className?: string
  title: string
  text: string
}

const ServiceOffering: React.FC<ServiceOfferingProps> = ({ title, text, className = '' }) => {
  return (
    <div className={className}>
      <div className="flex justify-center md:justify-start pb-4">
        {/* replace with image here */}
        <div className="bg-gray-300 dark:bg-gray-700 w-60 h-60 rounded-3xl"></div>
      </div>
      <div className="font-black text-2xl text-pink-600 pb-2 text-center md:text-left">
        {title}
      </div>
      <p className="text-center md:text-left">{text}</p>
    </div>
  )
}

export default ServiceOffering
