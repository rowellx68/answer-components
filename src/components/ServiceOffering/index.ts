import ServiceOffering from './ServiceOffering'
export default ServiceOffering
