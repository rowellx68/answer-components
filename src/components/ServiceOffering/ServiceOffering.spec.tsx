import React from 'react'
import renderer from 'react-test-renderer'

import ServiceOffering from './ServiceOffering'

test('render ServiceOffering correctly', () => {
  const element = renderer
    .create(<ServiceOffering title="Title" text="Text goes here" />)
    .toJSON()

  expect(element).toMatchSnapshot()
})
