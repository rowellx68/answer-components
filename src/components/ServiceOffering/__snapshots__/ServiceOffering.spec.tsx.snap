// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ServiceOffering correctly 1`] = `
<div
  className=""
>
  <div
    className="flex justify-center md:justify-start pb-4"
  >
    <div
      className="bg-gray-300 dark:bg-gray-700 w-60 h-60 rounded-3xl"
    />
  </div>
  <div
    className="font-black text-2xl text-pink-600 pb-2 text-center md:text-left"
  >
    Title
  </div>
  <p
    className="text-center md:text-left"
  >
    Text goes here
  </p>
</div>
`;
